src/EnergyPlus/SolarShading.cc

```diff
@@ -10018,6 +10019,29 @@ void WindowShadingManager(EnergyPlusData &state)
             }
         } // End of surface loop
     }
+    for (int enclosureNum = 1; enclosureNum <= state.dataViewFactor->NumOfSolarEnclosures; ++enclosureNum) {
+        state.dataHeatBal->EnclRadReCalc(enclosureNum) = false;
+        if (state.dataGlobal->BeginEnvrnFlag || state.dataGlobal->AnyConstrOverridesInModel ||
+            (state.dataWindowManager->inExtWindowModel->isExternalLibraryModel() && state.dataWindowManager->winOpticalModel->isSimplifiedModel())) {
+            state.dataHeatBal->EnclRadReCalc(enclosureNum) = true;
+            continue;
+        }
+        for (int const SurfNum : state.dataViewFactor->ZoneRadiantInfo(enclosureNum).SurfacePtr) {
+            if (state.dataConstruction->Construct(state.dataSurface->Surface(SurfNum).Construction).TCFlag == 1 ||
+                state.dataConstruction->Construct(state.dataSurface->Surface(SurfNum).Construction).WindowTypeEQL) {
+                state.dataHeatBal->EnclRadReCalc(enclosureNum) = true;
+                break;
+            }
+            bool surfShadingStatusChange =
```

Comment:
Is there a flag to skip these checks if there are no window shades or blinds or .... in the model?

Reply:
I added a check of if `state.dataGlobal->AndShadingControlInModel`. If not, `recalc` is always false and the following checks are bypassed.